### PR TITLE
Use standard bulk download/upload workflow for location rationalization

### PR DESCRIPTION
**Pull request by orangejenny**
PR into https://github.com/dimagi/commcare-hq/pull/26865

Screen Shot 2020-03-18 at 10 43 44 AM

### CODE REVIEW

File: custom/icds/location_rationalization/views.py [hunk 1]
```diff
@@ -1,120 +1,115 @@
...
     @property
     def page_context(self):
         context = super().page_context
-        context['form'] = self.form
+        context.update({
+            'bulk_upload': {
+                "download_url": reverse('download_location_rationalization', args=[self.domain]),
+                "adjective": _("locations"),
+                "plural_noun": _("location operations"),
+                "help_link": "TODO",
```
Comment (orangejenny) [inline custom/icds/location_rationalization/views.py]: @mkangia Have you started docs for this feature?

Comment (mkangia) [reply custom/icds/location_rationalization/views.py]: Not yet.

File: custom/icds/location_rationalization/views.py [hunk 2]
```diff
@@ -1,120 +1,115 @@
...
+            },
+        })
+        context.update({
+            'bulk_upload_form': get_bulk_upload_form(context),
```
Comment (mkangia) [inline custom/icds/location_rationalization/views.py]: Isn't it a bit odd that this is a not a bulk upload?

File: custom/icds/location_rationalization/views.py [hunk 3]
```diff
@@ -1,120 +1,115 @@
...
+            messages.error(request, _("Missing following columns in sheet: {columns}").format(
+                columns=", ".join(missing_headers)
+            ))
+            return False
+        return True
 
     @cached_property
     def _location_types(self):
-        location_types = [lt.code for lt in LocationType.objects.by_domain(self.domain)]
+        location_types = LocationType.objects.by_domain(self.domain)
```
Comment (mkangia) [inline custom/icds/location_rationalization/views.py]: this does not look right. it should be the codes only.

**Review by mkangia — APPROVED** [review]
looks good post the one change that i think is incorrect, possibly due to merge conflict

**Comment by mkangia** [comment]
This looks great, thanks for taking care of this. :)
I would have pulled my hair out thinking about changes for this.

File: custom/icds/location_rationalization/views.py (hunk 2 above)
Comment (orangejenny) [reply custom/icds/location_rationalization/views.py]: A little bit, although you are uploading a file, it just isn't updating anything on the server.

File: custom/icds/location_rationalization/views.py (hunk 3 above)
Comment (orangejenny) [reply custom/icds/location_rationalization/views.py]: Thank, yep, not intended. 6893fe6b31a

Comment (orangejenny) [reply custom/icds/location_rationalization/views.py]: This breaks the location type dropdown, which needs the type name. Any objection to me changing it back?

Comment (orangejenny) [reply custom/icds/location_rationalization/views.py]: Added 06525be14b5 to https://github.com/dimagi/commcare-hq/pull/26865, though I'd rather show the type names than just the codes.

Comment (mkangia) [reply custom/icds/location_rationalization/views.py]: Oh i see, ya i don't mind changing it back, just would need to change where its called as well.

**Pull request merged**